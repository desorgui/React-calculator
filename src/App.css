.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

.showMenu {
  display: flex;
  min-width: 300px;
  justify-content: flex-end;
  padding: 0;
}

.showMenu li {
  list-style: none;
  margin-right: 30px;
}

.showMenu li a {
  text-decoration: none;
  padding: 10px 0;
  transition: all linear .5s;
}

.showMenu li a:hover,
li .active,
li a:active {
  color: rgb(216, 149, 4);
  border-bottom: solid 2px rgb(216, 149, 4);
}

.home,
.quote {
  padding: 0 7%;
}

.home {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
}

.home .home-img {
  width: 50%;
  height: 100%;
  background-image: url('./42222.jpg');
  background-repeat: no-repeat;
  background-position: center;
  background-size: 100%;
}

.home .home-content {
  width: 50%;
  height: 100%;
  padding: 0 3rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.calculatorPage {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  background-image: url('./math-bg.jpg');
  background-size: 100% 100%;
}

.navBar {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  background-color: rgba(240, 248, 255, 0.555);
  -webkit-backdrop-filter: blur(10px);
  backdrop-filter: blur(10px);
  
}

.navBar .container {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 2rem;
}

.navBar img {
  height: 60px;
  width: 100px;
}

.quote {
  margin-top: 20vh;
  text-align: center;
}
