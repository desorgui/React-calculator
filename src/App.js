import React from 'react';
import './App.css';
import CalculatorPage from './components/pages/CalculatorPage';
import Navbar from './components/Navbar';
import Home from './components/pages/Home';
import Quote from './components/pages/Quote';
import { BrowserRouter, Route, Routes } from 'react-router-dom';

const App = () => {
  return (
    <>
      <BrowserRouter>
        <Navbar />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/quote" element={<Quote />} />
          <Route path="/calculator" element={<CalculatorPage />} />
        </Routes>
      </BrowserRouter>
    </>
  );
};

export default App;
