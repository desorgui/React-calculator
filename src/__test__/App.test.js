import { render, screen } from '@testing-library/react';
import App from '../App';

it('testing title', () => {
  render(<App />);
  const text = screen.getByText('Maths Magician');
  const li = screen.getAllByRole('linktest')
  expect(li.length).toBe(3)
  expect(text).toMatchSnapshot();
});
