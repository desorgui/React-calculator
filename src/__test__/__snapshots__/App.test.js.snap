// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing title 1`] = `
<h2>
  Maths Magician
</h2>
`;
